### DISCUSSION TITLE
command line

### DISCUSSION THREAD

**Discussion by simoncireneo** [sumatrapdfreader/sumatrapdf#3545] [Q&A]
I would like to know how to do something like this:

first creating a BAT file with this commands

open file1.cbr
save file1.cbr as file1.pdf

open file2.cbr
save file2.cbr as file2.pdf

open file3.cbr
save file3.cbr as file3.pdf

and so on..

thank you

**Comment by GitHubRulesOK** [comment]
SumatraPDF command line has open file and print file not save file as something else 

You can of course "Microsoft Print to PDF" (With or without Portprompting) or any other virtual image printer.

The result of Print rather than save can be inferior in several ways
      A .cbz of 3 x A4 images (png source) = 77 KB
=Print.pdf of 3 x A4 fuzzier jpeg lossy images  with artefacts = 1851 KB
OR A.pdf save as pdf with lossless quality = 78 KB

I suggested try sendkeys in your issue as its likely to be "leaner" and more accurate result

typically along the lines of
```
sumatrapdf a.cbz
sendkeys 1 2 "a.cbz" "^s{tab}p{enter}^Q"
```

**Reply by simoncireneo** [reply]
thank now I have to figure how to send keys. I dont know, now, how to do it.
anyway I thank you for your comment

**Reply by GitHubRulesOK** [reply]
the **sendkeys I suggest to download** is a utility as a single exe downloaded it is in a zip file with a plain text password you will see when you attempt to copy out the exe how to use password to extract.

If you write the now extracted folder location in the batch file it will run from there

`"c:\my folder where I put\sendkeys.exe" timer1value timer2value "application filename" "keyBoard keys"`

When you startSumatraPDF with a file the "application filename" will be the current tabbed file thus keep names simple I used "a.cbz" for my test.

If you need to save the currently viewed file the sequence will be `"^s{tab}p{enter}^Q"` where ^ means CTRL and {special} will be per the list of keys. Here I use `{tab}` to allow p for PDF and {enter} to accept same name, but you can use any key in name including folders once you know how it works. 

One limitation with sendkeys is it cannot warn you if there is an overwrite needed 
so if you always say Y or N you need to add that key entry for such a question.

**Comment by simoncireneo** [comment]
thank you for your help. I will try
